compiler/sigmatch.nim

```diff
@@ -2346,6 +2388,14 @@ proc matchesAux(c: PContext, n, nOrig: PNode, m: var TCandidate, marker: var Int
     formal = if formalLen > 1: m.callee.n[1].sym else: nil # current routine parameter
     container: PNode = nil # constructed container
 
+  # Check if the argument count match with the parameter count,
+  # if doesn't match, fail fast
+  # See https://github.com/nim-lang/RFCs/issues/402
+
```

Comment:
please add tests to ensure this works

things to consider:
* varags[int]
* varags[untyped]
* optional params
etc